libmscore/xmlreader.cpp

```diff
@@ -604,6 +604,8 @@ void XmlReader::removeConnector(const ConnectorInfoReader& cref)
             c = c->prev();
       while (c) {
             ConnectorInfoReader* next = c->next();
+            if (next)
+                  next->resetPrev();
```

Comment:
This change should fix the use-after-free issue but there seem to be other issues with connector readers, see [my commit](https://github.com/musescore/MuseScore/pull/4278/commits/76a22bd8315b9bfa884d5e1284eb2a16494f691f) option to fix this.

Reply:
I was simply trying a minimal change to cope with the bug: unfortunately, I really do not understand in deep detail the code of the connector readers.